src/vs/base/browser/ui/centered/centeredViewLayout.ts

```diff
@@ -3,60 +3,93 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { SplitView, Sizing, IView, Orientation, ISplitViewStyles } from 'vs/base/browser/ui/splitview/splitview';
+import { SplitView, Sizing, Orientation, ISplitViewStyles, IView as ISplitViewView } from 'vs/base/browser/ui/splitview/splitview';
 import { $ } from 'vs/base/browser/dom';
 import { Event } from 'vs/base/common/event';
+import { IView } from 'vs/base/browser/ui/grid/gridview';
 
 export class CenteredViewLayout {
 
 	private splitView: SplitView;
-	readonly element: HTMLElement;
-	private active: boolean;
+	private element: HTMLElement;
+	private height: number;
+	private style: ISplitViewStyles;
 
-	constructor(container: HTMLElement, view: IView) {
-		this.element = $('.centered-view-layout');
-		container.appendChild(this.element);
-
-		this.splitView = new SplitView(this.element, {
-			inverseAltBehavior: true,
-			orientation: Orientation.HORIZONTAL
-		});
-
-		this.splitView.addView(view, Sizing.Distribute);
+	constructor(private container: HTMLElement, private view: IView) {
+		this.container.appendChild(this.view.element);
 	}
 
-	layout(size: number): void {
-		this.splitView.layout(size);
+	layout(width: number, height: number): void {
+		this.height = height;
+		if (this.splitView) {
+			this.splitView.layout(width);
+		} else {
+			this.view.layout(width, height);
+		}
 	}
 
 	isActive(): boolean {
-		return this.active;
+		return !!this.splitView;
 	}
 
 	styles(style: ISplitViewStyles): void {
-		this.splitView.style(style);
+		this.style = style;
+		if (this.splitView) {
+			this.splitView.style(this.style);
+		}
+	}
+
+	resetView(view: IView): void {
+		this.view = view;
+		if (this.splitView) {
+			this.splitView.removeView(1);
+			this.splitView.addView(this.getView(), Sizing.Distribute, 1);
+		}
+	}
+
+	private getView(): ISplitViewView {
+		return {
+			element: this.view.element,
+			maximumSize: this.view.maximumWidth,
+			minimumSize: this.view.minimumWidth,
+			onDidChange: Event.None,
+			layout: size => this.view.layout(size, this.height)
+		};
 	}
 
 	activate(active: boolean): void {
-		this.active = active;
-		if (this.active) {
-			const emptyView = {
-				element: $('.'),
+		if (active) {
```

Comment:
You should still make a `if (active === !!this.splitview) { return; }`, so you don't get any NPEs.